src/implementations/sign/index.ts

```diff
@@ -1 +1,42 @@
-export * from "./sign";
+import { PrivateKeyOption } from "../../commands/shared";
+import { getPrivateKey } from "../utils/wallet";
+import { documentsInDirectory, readOpenAttestationFile, writeDocumentToDisk, isDir } from "../utils/disk";
+import { signDocument, defaultSigners } from "@govtechsg/oa-did-sign";
+import path from "path";
+import mkdirp from "mkdirp";
+
+export interface SignCommand extends PrivateKeyOption {
+  rawDocumentsPath: string;
+  outputDir: string;
+  publicKey: string;
+  algorithm: string;
+}
+
+export const supportedAlgorithms = Array.from(defaultSigners.keys());
+
+export const sign = async ({
+  rawDocumentsPath,
+  outputDir,
+  publicKey,
+  algorithm,
+  key,
+  keyFile,
+}: SignCommand): Promise<void> => {
+  const privateKey = await getPrivateKey({ key, keyFile });
+  if (!privateKey) throw new Error("Private key is not specified (use -k or -f to specify key)");
+  if (!isDir(outputDir)) throw new Error("output-dir must be a director");
```

Comment:
typo => `directory`

We create the output dir after checking it doesn't exist ?

Reply:
fuck that's silly